debian/nautilus.postinst

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+XDG_MIMEAPPS_FILE=/etc/xdg/mimeapps.list
+XDG_MIMEAPPS_HEADER="[Default Applications]"
+
+if [ "$1" = configure ]; then
+    # First non-empty line should be the required header, otherwise the file
+    # will be considered corrupt and will be removed to be filled later.
+    if  [ -f ${XDG_MIMEAPPS_FILE} ]; then
+        sed -i -e "/^[ \t]*$/d" ${XDG_MIMEAPPS_FILE}
+        if [ "$(head -n 1 ${XDG_MIMEAPPS_FILE})" != "${XDG_MIMEAPPS_HEADER}" ]; then
+            # Let's save a backup, not too be too agressive, just in case.
+            mv ${XDG_MIMEAPPS_FILE} ${XDG_MIMEAPPS_FILE}.bak
```

Comment:
I think the usual debian way to do this would be to make a copy to `${file}.dpkg-new`, make edits, then swap the current file to `${file}.dpkg-old` and the new file to `${file}` if the new file does not match the current file. This also makes sure you play nice with any hardlinks instead of appending to an existing file.

Reply:
Ah, right. That sounds better, will change it to do that

Reply:
That also guards against something happening in the middle of your edits and leaving the system with a broken original file.

Reply:
Yes, atomicity is good :)